### PR TITLE
Fix user visible check

### LINKED ISSUE #21206: User with limited visibility cannot view its own profile
### Description

A user with limited visibility cannot open the profile page and gets an HTTPNotFound error (404). It works if the user sets the visibility to public.

In `models\user\user.go` (`IsUserVisibleToViewer`) no check is made if the user in question is also the viewer, which should be allowed in my opinion.

As an admin, I can see my profile and all the other users profiles.

Log (user is called `foo_bar`):

```
 router: completed GET /foo_bar for 127.0.0.1:54370, 404 Not Found in 5.4ms @ user/profile.go:29(user.Profile)
```

### Gitea Version

1.17.2

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

2.20.1

### Operating System

Debian 10 Buster

### How are you running Gitea?

Binary using systemd service unit.

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #21206

If user and viewer are equal the method should return true.
Also the common organization check was wrong as `count` can never be less then 0.

### CODE REVIEW

**Comment by 6543** [comment]
please add a **this case** as **test**  
so regressions dont happen so likely

**Review by delvh — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

File: models/user/user.go [hunk 1]
```diff
@@ -1283,6 +1283,9 @@ func IsUserVisibleToViewer(ctx context.Context, u, viewer *User) bool {
 		if viewer == nil || viewer.IsRestricted {
 			return false
 		}
+		if viewer.ID == u.ID {
```
Comment (lunny) [inline models/user/user.go]: There is also the problem when u.Visibility is VisibleTypeLimited. So I think put it in line 1270 is better, just like I did in https://github.com/go-gitea/gitea/pull/21211/files

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
```
--- FAIL: TestFixtureGeneration (0.10s)

    fixture_test.go:31: 

        	Error Trace:	/drone/src/models/fixture_test.go:31

        	            				/drone/src/models/fixture_test.go:34

        	Error:      	Should be true

        	Test:       	TestFixtureGeneration

        	Messages:   	Differences detected for access.yml

FAIL
```

ci fail related, fixtures are now inconsistent

**Comment by wxiaoguang** [comment]
bot

**Pull request merged**